### PR TITLE
Create whitelist of conjunctions to be removed from search query befo…

### PR DESCRIPTION
**Pull request by nshumoogum**
…re calling azure search

### What

See title. Search API now strips out the following strings in course or institution search:

```
conjunctions = {
    "&",
    "and",
    "for",
    "in",
    "the",
    "with"
}
```

As well as any variation of upper and lower casing, case insensitive!

### How to review

Point service to an environment with enough data (e.g. pre-prod or prod) and check the number of results returned are heavily reduced.

Check unit tests still pass

### CODE REVIEW

File: CourseSearchAPIHttpTrigger/helper.py [hunk 1]
```diff
@@ -74,3 +85,34 @@ def group_courses_by_institution(search_results, counts, limit, offset):
...
+    if course != "":
+        course = remove_conjunctions(course)
+
+    if institution != "":
+        institution = remove_conjunctions(institution)
+
+    return course, institution
+
+
+conjunctions = {
```
Comment (philratcliffe) [inline CourseSearchAPIHttpTrigger/helper.py]: I don't see why this needs global scope.

File: CourseSearchAPIHttpTrigger/helper.py [hunk 2]
```diff
@@ -74,3 +85,34 @@ def group_courses_by_institution(search_results, counts, limit, offset):
...
+    "and",
+    "for",
+    "in",
+    "the",
+    "with"
+}
+
+
+def remove_conjunctions(searchable_field):
+    dict_of_string_parts = searchable_field.split(" ")
```
Comment (philratcliffe) [inline CourseSearchAPIHttpTrigger/helper.py]: But this isn't a dict 🙂

File: CourseSearchAPIHttpTrigger/tests/test_helper.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+class TestRemoveConjunctions(unittest.TestCase):
+    def test_when_searchable_field_is_empty_string(self):
+        
+        output_result = remove_conjunctions("")
+        self.assertEqual("", output_result)
+
+    def test_when_searchable_field_has_ampersand_conjunction(self):
+        input_field = "Mathematics & Physics"
+        expected_result = "Mathematics Physics"
+
```
Comment (philratcliffe) [inline CourseSearchAPIHttpTrigger/tests/test_helper.py]: Trailing whitespace here and in several other places in this module. Black will fix.

File: CourseSearchAPIHttpTrigger/tests/test_helper.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        
+        output_result = remove_conjunctions(input_field)
+        self.assertEqual(expected_result, output_result)
+
+    def test_when_searchable_field_has_uppercase_conjunction(self):
+        input_field = "Mathematics WITH Physics"
+        expected_result = "Mathematics Physics"
+        
+        output_result = remove_conjunctions(input_field)
+        self.assertEqual(expected_result, output_result)
```
Comment (philratcliffe) [inline CourseSearchAPIHttpTrigger/tests/test_helper.py]: No final newline.

**Review by philratcliffe — COMMENTED** [review]
LGTM - added a few comments.

File: CourseSearchAPIHttpTrigger/helper.py (hunk 2 above)
Comment (nshumoogum) [reply CourseSearchAPIHttpTrigger/helper.py]: It was and then changed what the variable was doing without changing the name oops

File: CourseSearchAPIHttpTrigger/helper.py [hunk 5]
```diff
@@ -97,21 +97,20 @@ def remove_conjunctions_from_searchable_fields(course, institution):
...
-    "for",
-    "in",
-    "the",
-    "with"
-}
-
-
 def remove_conjunctions(searchable_field):
-    dict_of_string_parts = searchable_field.split(" ")
+    conjunctions = {
```
Comment (philratcliffe) [inline CourseSearchAPIHttpTrigger/helper.py]: Yep, better at that scope 👍

**Review by philratcliffe — APPROVED** [review]
Looks good

**Pull request merged**